src/client.ts

```diff
@@ -98,48 +100,67 @@ export class GenesysCloudWebrtcSdk extends (EventEmitter as { new(): StrictEvent
     return !this._config.accessToken;
   }
 
-  constructor (options: ISdkConstructOptions) {
+  constructor (options: ISdkConfig) {
     super();
 
+    options.media = options.media || {};
+    options.defaults = options.defaults || {};
+
     const errorMsg = validateOptions(options);
     if (errorMsg) {
       throw new SdkError(SdkErrorTypes.invalid_options, errorMsg);
     }
 
     this._config = {
+      environment: options.environment,
       accessToken: options.accessToken,
+      wsHost: options.wsHost,
+      organizationId: options.organizationId,
       autoConnectSessions: options.autoConnectSessions !== false, // default true
-      customIceServersConfig: options.iceServers,
-      defaultAudioElement: options.defaultAudioElement,
-      defaultAudioStream: options.defaultAudioStream,
-      defaultVideoElement: options.defaultVideoElement,
-      defaultVideoDeviceId: options.defaultVideoDeviceId || null,
-      defaultAudioDeviceId: options.defaultAudioDeviceId || null,
-      defaultOutputDeviceId: options.defaultOutputDeviceId || null,
-      disableAutoAnswer: options.disableAutoAnswer || false, // default false
-      environment: options.environment,
-      iceTransportPolicy: options.iceTransportPolicy || 'all',
       logLevel: options.logLevel || 'info',
+      disableAutoAnswer: options.disableAutoAnswer || false, // default false
       optOutOfTelemetry: options.optOutOfTelemetry || false, // default false
       allowedSessionTypes: options.allowedSessionTypes || Object.values(SessionTypes),
-      wsHost: options.wsHost
+
+      /* media related config */
+      media: {
+        monitorMicVolume: !!options.media.monitorMicVolume, // default to false
```

Comment:
I was thinking about adding an option to _not_ try the "hail mary" media retry. Maybe that would be overkill?